src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -54,7 +54,20 @@ let SerializeAddon: typeof SerializeAddonType;
 export class XtermTerminal extends DisposableStore implements IXtermTerminal, IInternalXtermTerminal {
 	/** The raw xterm.js instance */
 	readonly raw: RawXtermTerminal;
+	private _bufferLines: string[] = [];
```

Comment:
I don't think we want to cache this, otherwise the memory will be held. If we do want to cache line strings, I remember thinking we could do it within xterm.js so that the search addon could benefit as well.

Reply:
created an issue and now since we'll be checking in reverse order I don't think caching is necessary